src/command/client/sync/login.rs

```diff
@@ -40,8 +43,25 @@ impl Cmd {
         }
 
         let username = or_user_input(&self.username, "username");
-        let key = or_user_input(&self.key, "encryption key");
+        let key = or_user_input(&self.key, "encryption key [blank to use existing key file]");
         let password = self.password.clone().unwrap_or_else(read_user_password);
+
+        let key_path = settings.key_path.as_str();
+        if key.is_empty() {
+            let mut file = match File::open(key_path).await {
+                Err(_e) => bail!("key file couldn't be opened, key must be specified"),
+                Ok(f) => f,
+            };
+            let mut bytes = Vec::new();
+            file.read_to_end(&mut bytes).await?;
+            if bytes.is_empty() {
+                bail!("key file empty, key must be specified");
+            }
```

Comment:
```suggestion
            let bytes = tokio::fs::read(key_path)
                .await
                .context("key file couldn't be opened, key must be specified")?;

            if bytes.is_empty() {
                bail!("key file empty, key must be specified");
            }
```
(the `context` here needs `eyre:: WrapErr` to be imported)

Reply:
Thanks! I have written very little rust, so not surprised it could be nicer. I applied the patch and included the import.